### PR TITLE
cleanup queues for non-existent pools and non-existent tasks

### LINKED ISSUE #2432: Add task queue garbage collection
Add queue task garbage collection in case tasks records are deleted from task table manually. Then delete task queues when tasks do not exist

### PR DESCRIPTION
**Pull request by stishkin**
closes #2432

### CODE REVIEW

**Review by Porges — APPROVED** [review]

File: src/ApiService/ApiService/onefuzzlib/Queue.cs [hunk 1]
```diff
@@ -30,6 +34,12 @@ public class Queue : IQueue {
         _log = log;
     }
 
+    public async IAsyncEnumerable<QueueItem> ListQueues(StorageType storageType) {
+        var queueServiceClient = await GetQueueClientService(storageType);
+        await foreach (var q in queueServiceClient.GetQueuesAsync()) {
```
Comment (Porges) [inline src/ApiService/ApiService/onefuzzlib/Queue.cs:39]: this can return `queueServiceClient.GetQueuesAsync()` directly rather than yielding each item

Comment (stishkin) [reply src/ApiService/ApiService/onefuzzlib/Queue.cs:39]: like this ?
image

Comment (stishkin) [reply src/ApiService/ApiService/onefuzzlib/Queue.cs:39]: I think if you suggestion works - we can do this in ORM as well

**Review by stishkin — COMMENTED** [review]
..

**Pull request merged**